tbx/core/blocks.py

```diff
@@ -320,27 +328,59 @@ class Meta:
         template = "patterns/molecules/streamfield/blocks/work_chooser_block.html"
 
 
-class StoryBlock(StreamBlock):
-    h2 = CharBlock(
+class PhotoCollageStructValue(blocks.StructValue):
+    def url(self):
+        return self.get("page").url
+
+    def button_text(self):
+        if self.get("page") and not self.get("link_text"):
+            return self.get("page").title
+        return self.get("link_text")
+
+
+class PhotoCollageBlock(blocks.StructBlock):
+    title = blocks.CharBlock(max_length=255)
+    intro = blocks.TextBlock(label="Introduction")
+    page = blocks.PageChooserBlock()
+    link_text = blocks.CharBlock(
+        required=False,
+        max_length=55,
+        help_text="Leave blank to use the page's own title",
+    )
+    images = blocks.ListBlock(
+        BaseImageBlock(label="Photo"),
+        min_num=6,
+        max_num=6,
+        label="Photos",
+    )
```

Comment:
**Suggestion**: Consider also adding a default. If editors always have to add 6, it may be useful to see the fields straight away:

```suggestion
       default=[{"image": None, "alt_text": ""}] * 6,
    )
```

Maybe the UX of this is unusual though. What do you think?

Reply:
Thanks! I think it's a good idea -- the editor is immediately made aware of the fact that they need to add 6 photos, and they don't have to keep clicking "Add photo" multiple times!